app/packs/src/apps/mydb/elements/details/samples/propertiesTab/SampleForm.js

```diff
@@ -726,42 +659,46 @@ export default class SampleForm extends React.Component {
       <Form.Group className="flex-grow-1">
         <Form.Label>Amount</Form.Label>
         {sample.isMethodDisabled('amount_value') ? (
-          <FormControl type="text" disabled defaultValue="***" readOnly />
+          <Form.Control type="text" disabled defaultValue="***" readOnly />
         ) : (
-          <div className='d-flex gap-2'>
+          <div className="d-flex gap-2">
             {this.numInput(sample, 'amount_g', 'g', ['m', 'n', 'u'], 4, null, 'massMgInput', isDisabled, '')}
             {this.numInput(sample, 'amount_l', 'l', ['m', 'u', 'n'], 5, null, 'l', isDisabled, '', volumeBlocked)}
             {this.numInput(sample, 'amount_mol', 'mol', ['m', 'n'], 4, null, 'amountInput', isDisabled, '')}
-            {sample.contains_residues && this.attachedAmountInput(sample)}
           </div>
         )}
       </Form.Group>
     );
   }
 
-  densityMolarityInput(sample, isPolymer, molarityBlocked, densityBlocked) {
+  densityMolarityInput(sample) {
+    const { densityMolarity } = this.state;
+    const isPolymer = (sample.molfile || '').indexOf(' R# ') !== -1;
+    const isDisabled = !sample.can_update;
+    const polyDisabled = isPolymer || isDisabled;
+
     return (
       <>
         <ButtonGroup className="mb-2">
           <Button
             onClick={() => this.setState({ densityMolarity: 'density' })}
-            variant={this.state.densityMolarity === 'density' ? 'info' : 'light'}
+            variant={densityMolarity === 'density' ? 'info' : 'light'}
             size="xxsm"
           >
             Density
           </Button>
           <Button
             onClick={() => this.setState({ densityMolarity: 'molarity' })}
-            variant={this.state.densityMolarity === 'molarity' ? 'info' : 'light'}
+            variant={densityMolarity === 'molarity' ? 'info' : 'light'}
```

Comment:
We have dedicated Buttons for ButtonGroups with toggle functionality in another branch...